kolibri/plugins/coach/assets/src/composables/useQuizResources.js

```diff
@@ -0,0 +1,138 @@
+    return ContentNodeResource.fetchDescendantsAssessments(topicIds)
+      .then(({ data: topicsWithDescendantCounts }) => {
+        //console.log(topicsWithDescendantCounts);
+        const childrenWithAnnotatedTopics = get(_resources)
+          .map(node => {
+            // We'll map so that the topics are updated in place with the num_assessments, others
+            // are left as-is
+            if (node.kind === ContentNodeKinds.TOPIC) {
+              const topic = topicsWithDescendantCounts.find(t => t.id === node.id);
+              if (topic) {
+                node.num_assessments = topic.num_assessments;
+              }
+              if (!validateObject(node, QuizExercise)) {
+                console.warn('Topic node was not a valid QuizExercise after annotation:', node);
+              }
+            }
+            return node;
+          })
+          .filter(node => {
+            // Only keep topics which have assessments in them to begin with
+            if (node.kind === ContentNodeKinds.TOPIC) {
+              if (node.num_assessments > 0) {
+                return true;
+              } else {
+                return false;
+              }
+            }
+            return true; // Not a topic, so must be an Exercise due to API params
+          });
+        set(_resources, childrenWithAnnotatedTopics);
+      })
+      .catch(e => {
+        // TODO Work out best UX for this situation -- it may depend on if we're fetching more
+        // or the initial list of contents
+        console.error(e);
+      });
+  }
+
+  /**
+   *  @affects _resources - Sets the _resources to the results of the fetchTree call
+   *  @affects _loading
+   *  @returns {Promise<null>} - A promise that resolves when the annotations have been made and
+   */
+  async function fetchQuizResources() {
+    set(_loading, true);
+    return fetchTree().then(async results => {
+      setResources(results);
+      return annotateTopicsWithDescendantCounts(
+        results.filter(({ kind }) => kind === ContentNodeKinds.TOPIC).map(topic => topic.id)
+      ).then(() => set(_loading, false));
+    });
+  }
+
+  /**
+   *  @affects _resources - Appends the results of the fetchMore call to the _resources
+   *    and annotates any new topics with descendant counts
+   *  @affects _loading - fetchMore & annotateTopicsWithDescendantCounts update the loading states
+   */
+  async function fetchMoreQuizResources() {
+    set(_loading, true);
+    return fetchMore().then(async results => {
+      console.log(results);
+      set(_resources, [...get(_resources), ...results]);
+      return annotateTopicsWithDescendantCounts(
+        results.filter(({ kind }) => kind === ContentNodeKinds.TOPIC).map(topic => topic.id)
+      ).then(() => set(_loading, false));
+    });
+  }
+
+  /** @returns {Boolean} Whether the given node should be displayed with a checkbox
+   * currently passes when the node is an Exercise or if it is a topic with fewer than 20
```

Comment:
Do we have explanatory text for this anywhere? (if so, probably good to store the value as a constant to feed into the text, and here, in case we decide to change it).

Reply:
I'm not 100% sure how the question references the highlighted lines.

I removed the condition around topic selection altogether so for now we'll only deal w/ allowing Exercises to be selected.

Given the complexity in identifying whether or not we have all of the topics descendants loaded by way of fetchTree -- namely with consideration for the fact that we may not have all of the first level's children if we `hasMore` -- this seems like the best way to go.

Reply:
Hrm, that's a little bit of a step back - as it can be helpful to select an entire topic full of exercises to create a quiz.

We had previously discussed only allowing topics that had no topic descendants to be selected - but I guess knowledge of that does slightly depend on whether you have fetched the grandchildren of the current topic or not.

Reply:
We discussed this on slack and came up with:

> the limit is you can select a folder if we fetched every child in one go (an implicit size limit of 12) and every child is not a folder

Will be filed as a follow up issue.